### PR TITLE
[ST] Add a way to simulate tls-external auth client producer and consumer

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR adds methods into client utils, so that it can simulate an end user providing its own certificates to test producing and consuming messages as operator does not generate ones for authentication type tls-external.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST#testTlsExternalUser --env=STRIMZI_RBAC_SCOPE=NAMESPACE

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST#testTlsExternalUser --env=STRIMZI_RBAC_SCOPE=NAMESPACE

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST#testTlsExternalUser --env=STRIMZI_RBAC_SCOPE=NAMESPACE

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --testcase=UserST#testTlsExternalUser --env=STRIMZI_RBAC_SCOPE=NAMESPACE

File: systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class OpenSsl {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java:22]: Adding a JavaDoc to this class would be great :) explaining the main purpose of it and what is provides...

File: systemtest/src/main/java/io/strimzi/systemtest/security/SystemTestCertManager.java [hunk 2]
```diff
@@ -169,6 +170,21 @@ private static File exportCertsToPemFile(SystemTestCertAndKey... certs) throws I
         return certFile;
     }
 
+    public static File exportCaDataToFile(String caData, String prefix, String suffix) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/security/SystemTestCertManager.java:173]: JavaDoc? :)

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 3]
```diff
@@ -340,123 +344,100 @@ void testCreatingUsersWithSecretPrefix(ExtensionContext extensionContext) {
...
-        final String userName = testStorage.getKafkaUsername();
-        final String topicName = testStorage.getTopicName();
-
-        final AclRule aclRule = new AclRuleBuilder()
-            .withNewAclRuleTopicResource()
-                .withName(topicName)
-            .endAclRuleTopicResource()
-            .withOperations(AclOperation.WRITE, AclOperation.DESCRIBE)
-            .build();
+        String consumerGroupName = ClientUtils.generateRandomConsumerGroup();
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:349]: Don't we generate a consumer group in the TestStorage class?

**Review by see-quick — APPROVED** [review]
Assuming the test passes. LGTM.

**Review by henryZrncik — APPROVED** [review]
good job LGTM 👍

File: systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java (hunk 1 above)
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java:22]: Will do. Thanks!

File: systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        public String executeAndReturnOnSuccess(boolean failOnNonZeroOutput) {
+
+            Path commandOutput = null;
+            try {
+                commandOutput = Files.createTempFile("openssl-command-output-", ".txt");
+
+                pb.redirectErrorStream(true)
+                    .redirectOutput(commandOutput.toFile());
+
+                LOGGER.info("Running command: {}", pb.command());
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/security/OpenSsl.java:71]: Debug maybe?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 3 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:349]: Unfortunately no, we only get consumerName.

**Pull request merged**